### PR TITLE
Redux reducers for Add and Delete

### PR DESCRIPTION
**Pull request by RafaelEchart**
## Milestone review:

- Added redux cycle to Add and Remove books from the global state managed by Redux.
- Added little details for the user experience such as cleaning the Name, author, and Genre after the book is added.
- Rendering is connected to the global state managed by Redux
- Added notification when a Book is successfully added and when the inputs need correction.

### Additional feature:

- When a new book is added a random number between 0 and 100 is generated and added in the book information, this is done so we can see how the progress circle changes when passing different % (This will change in the next project)

### SETUP INSTRUCTIONS:

- npm install
- npm run start

### CODE REVIEW

File: src/components/BooksPage/index.js [hunk 1]
```diff
@@ -1,46 +1,29 @@
...
             <OneBook oneBook={book} key={book.id} />
-    ))}
+      ))
+      : <NoBooksMessage />}
 
     <NewBook />
 
       </div>
 
       </>
```
Comment (TedLivist) [inline src/components/BooksPage/index.js]: - [ ] I would recommend indenting your code properly so that your code looks more readable and neat.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @RafaelEchart,

Good job so far!
✔️ Your UI looks very nice
✔️ Your added idea for displaying messages is quite awesome 
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [x] From the image below you seem to have edited or used a different eslint file for your project. Kindly use the linter configuration by Microverse.

  eckhart-linter

- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BooksPage/OneBook.js [hunk 2]
```diff
@@ -1,49 +1,62 @@
...
+        </div>
+      </div>
+    </div>
 
   );
 };
-
+OneBook.propTypes = {
+  oneBook: PropTypes.element.isRequired,
+};
```
Comment (ArthurGC) [inline src/components/BooksPage/OneBook.js]: - [OPTIONAL] There is a console error about props. It seems the type of prop is wrong.

image

**Review by ArthurGC — APPROVED** [review]
# Approved project!  💯 ✅ 🥳 🎈 🎉
Hi  @RafaelEchart 

Amazing job implementing suggestions from the previous reviewer! 💯 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/components/BooksPage/OneBook.js (hunk 2 above)
Comment (RafaelEchart) [reply src/components/BooksPage/OneBook.js]: Thanks, I just fixed this!

**Pull request merged**